### PR TITLE
enhance mosn smooth upgrade for issue #1397

### PR DESCRIPTION
**Pull request by alpha-baby**
### Issues associated with this PR

Your PR should present related issues you want to solve.

### Solutions
You should show your solutions about the issues in your PR, including the overall solutions, details and the changes. At this time, Chinese is allowed to describe these.

### UT result
Unit Test is needed if the code is changed, your unit test should cover boundary cases, corner cases, and some exceptional cases. And you need to show the UT result.

### Benchmark
If your code involves the processing of every request, you should give the Benchmark Result.

### Code Style
+ Make sure `Goimports` has run
+ Show `Golint` result

### CODE REVIEW

File: pkg/filter/network/connectionmanager/deprecated.go [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
 import (
+	"fmt"
 	"mosn.io/api"
 	"mosn.io/mosn/pkg/config/v2"
 	"mosn.io/mosn/pkg/log"
 )
 
 func init() {
 	api.RegisterNetwork(v2.CONNECTION_MANAGER, CreateProxyFactory)
+	fmt.Printf("registry CONNECTION_MANAGER")
```
Comment (champly) [inline pkg/filter/network/connectionmanager/deprecated.go:12]: need this code?

File: pkg/types/config.go [hunk 2]
```diff
@@ -24,7 +24,7 @@ import (
 )
 
 var (
-	MosnBasePath = string(os.PathSeparator) + "home" + string(os.PathSeparator) +
+	MosnBasePath = "/Users/fujianhao3/Desktop/bak/mosn/.mydevCachefile/bak" + string(os.PathSeparator) + "home" + string(os.PathSeparator) +
```
Comment (champly) [inline pkg/types/config.go:27]: Remove prefix path

File: pkg/mosn/starter.go [hunk 3]
```diff
@@ -91,6 +91,17 @@ func NewMosn(c *v2.MOSNConfig) *Mosn {
 		store.SetMosnState(store.Active_Reconfiguring)
 		// parse MOSNConfig again
 		c = configmanager.Load(configmanager.GetConfigPath())
+		// inherit old mosn config
+		oldMosnConfig, err := server.GetInheritConfig()
+		if err != nil {
+			log.StartLogger.Fatalf("[mosn] [NewMosn] GetInheritConfig failed, exit")
+		}
+		log.StartLogger.Debugf("[mosn] [NewMosn] old mosn config: %v", oldMosnConfig)
+		if c.InheritOldMosnconfig {
```
Comment (nejisama) [inline pkg/mosn/starter.go:100]: if no InheritOldMosnconfig, do not needs to call server.GetInheritConfig()

File: pkg/server/handler.go [hunk 4]
```diff
@@ -20,8 +20,11 @@ package server
 import (
 	"container/list"
 	"context"
+	"encoding/json"
 	"errors"
 	"fmt"
+	"io"
+	v2 "mosn.io/mosn/pkg/config/v2"
```
Comment (nejisama) [inline pkg/server/handler.go:27]: go imports format

File: pkg/server/reconfigure.go [hunk 5]
```diff
@@ -82,6 +82,9 @@ func reconfigure(start bool) {
 	if listenSockConn, err = sendInheritListeners(); err != nil {
 		return
 	}
+	if err = SendInheritConfig(); err != nil {
```
Comment (nejisama) [inline pkg/server/reconfigure.go:85]: maybe also needs to check InheritOldMosnconfig; if not , do not do this

File: pkg/server/handler.go [hunk 6]
```diff
@@ -813,6 +815,43 @@ func sendInheritListeners() (net.Conn, error) {
 	return uc, nil
 }
 
+// SendInheritConfig send to new mosn using uinx dowmain socket
+func SendInheritConfig() error {
+	configData, err := configmanager.InheritMosnconfig()
```
Comment (nejisama) [inline pkg/server/handler.go:820]: get config after dial success

**Comment by nejisama** [comment]
88 commits?

**Comment by nejisama** [comment]
#1459 instead of this one

**Pull request closed without merging**